cmake/platform.cmake

```diff
@@ -141,23 +151,34 @@ if(OMR_HOST_OS STREQUAL "win")
 		set(linker_common "${linker_common} /SAFESEH")
 	endif()
 
-	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${linker_common} /INCREMENTAL:NO /NOLOGO /LARGEADDRESSAWARE wsetargv.obj")
+	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${linker_common} /LARGEADDRESSAWARE wsetargv.obj")
 	if(OMR_ENV_DATA64)
 		#TODO: makefile has this but it seems to break linker
 		#set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /NODEFAULTLIB:MSVCRTD")
 	endif()
 
+
+	# Make sure we are building without incremental linking
+	omr_remove_option(CMAKE_EXE_LINKER_FLAGS "/INCREMENTAL")
+	omr_remove_option(CMAKE_SHARED_LINKER_FLAGS "/INCREMENTAL")
+	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /INCREMENTAL:NO /NOLOGO")
 	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /INCREMENTAL:NO /NOLOGO")
+	foreach(build_type IN LISTS CMAKE_CONFIGURATION_TYPES)
+        string(TOUPPER "${build_type}" build_type)
+		omr_remove_option("CMAKE_EXE_LINKER_FLAGS_${build_type}" "/INCREMENTAL")
+		omr_remove_option("CMAKE_SHARED_LINKER_FLAGS_${build_type}" "/INCREMENTAL")
+	endforeach()
+
 	if(OMR_ENV_DATA64)
 		set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -entry:_DllMainCRTStartup")
 	else()
 		set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -entry:_DllMainCRTStartup@12")
 	endif()
 
 	#strip out exception handling flags (added by default by cmake)
-	string(REPLACE "/EHsc" "" filtered_cxx_flags ${CMAKE_CXX_FLAGS})
-	string(REPLACE "/GR" "" filtered_cxx_flags ${filtered_cxx_flags})
-	set(CMAKE_CXX_FLAGS "${filtered_cxx_flags} ${common_flags}")
+	omr_remove_option(CMAKE_CXX_FLAGS "/EHsc")
+	omr_remove_option(CMAKE_CXX_FLAGS "/GR")
```

Comment:
I dislike mandating language feature flags, some uses might need these things. I'd rather these flags be configured externally.  If it's needed for CI,  can we (un)set these flags through a cmake cache or the environment?